lib/runner.js

```diff
@@ -142,52 +159,72 @@ Runner.prototype.onExit = function onExit(code, signal) {
 
   debug('Exit of %s with status %s stopped? %s', this, status, this.stopped);
 
+  this.child.exited = true;
+  this.child.exitStatus = code || signal;
```

Comment:
`signal || code`: status code 0 is false, so on normal exit `0` the exitStatus in your version will be `undefined`.